pkg/querier/queryrange/results_cache.go

```diff
@@ -37,8 +37,14 @@ type ResultsCacheConfig struct {
 // RegisterFlags registers flags.
 func (cfg *ResultsCacheConfig) RegisterFlags(f *flag.FlagSet) {
 	cfg.CacheConfig.RegisterFlagsWithPrefix("frontend.", "", f)
+
 	f.DurationVar(&cfg.MaxCacheFreshness, "frontend.max-cache-freshness", 1*time.Minute, "Most recent allowed cacheable result, to prevent caching very recent results that might still be in flux.")
-	f.DurationVar(&cfg.SplitInterval, "frontend.cache-split-interval", 24*time.Hour, "The maximum interval expected for each request, results will be cached per single interval.")
+	f.DurationVar(&cfg.SplitInterval, "frontend.cache-split-interval", 0, "Deprecated: The maximum interval expected for each request, results will be cached per single interval. This flag has been deprecated and this behavior is now determined by querier.split-queries-by-interval")
```

Comment:
I would suggest to remove `SplitInterval` at all. The change introduced in this PR is a breaking change anyway, and removing this config option would allow to have a cleaner `config-file-reference.md` as well as signal the change to an user which upgrades without reading the `CHANGELOG.md` (because the config parsing will fail).

What's your take?

Reply:
I opted to not have this be a breaking change, but instead to provide a deprecation warning but still use the old config if specified.

Reply:
Yes this is not a breaking change, yet.